### PR TITLE
Add High Number of Leader Changes test

### PR DESCRIPTION
**Pull request by devopsjonas**

### CODE REVIEW

File: alerts/monquorum.libsonnet [hunk 1]
```diff
@@ -23,7 +23,7 @@
           {
             alert: 'CephMonHighNumberOfLeaderChanges',
             expr: |||
-              rate(ceph_mon_num_elections{%(cephExporterSelector)s}[15m]) > 3
+              rate(ceph_mon_num_elections{%(cephExporterSelector)s}[10m]) > 0
```
Comment (shtripat) [inline alerts/monquorum.libsonnet]: I feel here you are changing the rules. I would expect these changes to be part of separate PR with proper comments why we need this change.

File: test_alerts.yaml [hunk 2]
```diff
@@ -2,6 +2,23 @@ rule_files:
...
+    alert_rule_test:
+      - alertname: CephMonHighNumberOfLeaderChanges
+        eval_time: 11m
+        exp_alerts:
+         - exp_labels:
+             severity: warning
+             job: rook-ceph-mgr
+           exp_annotations:
+             message: 'Storage Cluster has seen many leader changes recently.'
+             description: 'Ceph Monitor "rook-ceph-mgr": instance  has seen 0.016666666666666666 leader changes recently.'
```
Comment (shtripat) [inline test_alerts.yaml]: Is 0.01666... many leader changes? Lets try to put more realistic sample data may be :)

File: alerts/monquorum.libsonnet (hunk 1 above)
Comment (devopsjonas) [reply alerts/monquorum.libsonnet]: Can we make this into this PR as tests actually add more context and show actual examples where the current alert configuration fail. WDYT?

File: test_alerts.yaml (hunk 2 above)
Comment (devopsjonas) [reply test_alerts.yaml]: This shows leader changes per second, if you multiply it by 60, which is ~1 leader change per minute.

**Comment by devopsjonas** [comment]
@shtripat I think we should change the alert rule to be something like 1 leader change per minute within last 5 minutes.

so the expression would be `rate(ceph_mon_num_elections{job...}[5m]) * 60 > 1`, WDYT?

And `for` label should be close to 0.

Also maybe instead of `1`, we should do `0.95`, as currenlty ` 0.01666` changes per second * 60 =  0.9996, so alert wouldn't fire due to rounding and rate approximations.,

File: alerts/monquorum.libsonnet (hunk 1 above)
Comment (shtripat) [reply alerts/monquorum.libsonnet]: Ack

**Comment by shtripat** [comment]
> @shtripat I think we should change the alert rule to be something like 1 leader change per minute within last 5 minutes.
> 
> so the expression would be `rate(ceph_mon_num_elections{job...}[5m]) * 60 > 1`, WDYT?
> 
> And `for` label should be close to 0.
> 
> Also maybe instead of `1`, we should do `0.95`, as currenlty ` 0.01666` changes per second * 60 = 0.9996, so alert wouldn't fire due to rounding and rate approximations.,

Looks fine.

**Comment by shtripat** [comment]
Please rebase and see if build errors are gone now.

**Comment by devopsjonas** [comment]
@shtripat done, I've updated alert.

**Review by shtripat — COMMENTED** [review]
Looks good to me.
@umangachapagain please review once

File: config.libsonnet [hunk 3]
```diff
@@ -12,7 +12,7 @@
     clusterVersionAlertTime: '10m',
     clusterUtilizationAlertTime: '5m',
     monQuorumAlertTime: '15m',
-    monQuorumLeaderChangesAlertTime: '30m',
+    monQuorumLeaderChangesAlertTime: '0s',
```
Comment (umangachapagain) [inline config.libsonnet]: I am assuming this is `0s` for getting instant alerts.
But, what about cases where cluster is getting initialized and mon experiences frequent changes in leader for some reason?
I would say this should be set to something `>= 5m`.

Comment (devopsjonas) [reply config.libsonnet]: You are right, I didn't think of cluster creating. Changed to 5m. Please take a look

**Review by shtripat — APPROVED** [review]
LGTM

**Review by umangachapagain — APPROVED** [review]

File: alerts/monquorum.libsonnet [hunk 4]
```diff
@@ -23,15 +23,15 @@
...
+              rate(ceph_mon_num_elections{%(cephExporterSelector)s}[5m]) * 60 > 0.95
             ||| % $._config,
             'for': $._config.monQuorumLeaderChangesAlertTime,
             labels: {
               severity: 'warning',
             },
             annotations: {
               message: 'Storage Cluster has seen many leader changes recently.',
-              description: 'Ceph Monitor "{{ $labels.job }}": instance {{ $labels.instance }} has seen {{ $value }} leader changes recently.',
+              description: 'Ceph Monitor "{{ $labels.job }}": instance {{ $labels.instance }} has seen {{ $value }} leader changes per minute recently.',
```
Comment (cloudbehl) [inline alerts/monquorum.libsonnet:34]: can be change `per minute recently` to `frequently`. As per minute is a time duration we might not want to fix that value in message.
